name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/FidelityPDFExtractor.java

```diff
@@ -0,0 +1,216 @@
+
+                        .section("fee") //
+                        .optional() //
+                        .match(".*Activity Assessmen\\s?t\\s+F\\s?e\\s?e\\s+(?<fee>[\\.,\\d]+)")
+                        .assign((t, v) -> {
+                            t.getPortfolioTransaction()
+                                        .addUnit(new Unit(Unit.Type.FEE,
+                                                        Money.of(asCurrencyCode(asCurrencyCode("USD")),
+                                                                            asAmount(v.get("fee")))));
+                        })
+
+                        .section("shares") //
+                        .match("^\\s+(?<shares>[\\.,\\d]+)\\s.*$") //
+                        .assign((t, v) -> t.setShares(asShares(v.get("shares"))))
+
+                        .wrap(BuySellEntryItem::new));
+    }
+
+    private void addBuySellTransaction()
+    {
+        // line must start with this!
+        DocumentType type = new DocumentType("SECURITY DESCRIPTION");
+        this.addDocumentTyp(type);
+
+        Block block = new Block(
+                        "CUSTOMER NO\\.\\s+PARTICIPANT ID\\. TYPE REG\\.REP\\. TRADE DATE SETTLEMENT DATE TRANS NO\\. CUSIP NO\\. ORIG\\.");
+        type.addBlock(block);
+        block.set(new Transaction<BuySellEntry>()
+
+                        .subject(() -> {
+                            BuySellEntry entry = new BuySellEntry();
+                            entry.setType(PortfolioTransaction.Type.SELL);
+                            return entry;
+                        })
+                        /* @formatter:off
+                         * CUSTOMER NO.   PARTICIPANT ID. TYPE REG.REP. TRADE DATE SETTLEMENT DATE TRANS NO. CUSIP NO. ORIG.
+                         * I00123456 1 WI# 12-12-23 12-14-23 K9T1Q9 11135F101
+                         * YOU SOLD 14 AT 173.1100
+                         *       EXPLANATION OF PROCEEDS
+                         * SECURITY DESCRIPTION SYMBOL: AAPL Sale Proceeds     $2,423.54
+                         * APPLE INC
+                         * DETAILS: Total Fees          $0.13
+                         * Sale Date: DEC/12/2023
+                         * Proceeds Available: DEC/14/2023
+                         * Plan Type: COMPANY STOCK PLAN
+                         * Net Cash Proceeds¹     -$2,423.41
+                         * @formatter:on
+                         */
+
+                        .section("date") //
+                        .match("^Sale Date: (?<date>[\\w]{3}\\/[\\d]{2}\\/[\\d]{4})$")
+                        .assign((t, v) -> {
+                            String date = v.get("date");
+                            // convert month to camel case
+                            if (Character.isLetter(date.charAt(0)))
+                            {
+                                date = date.charAt(0) + date.substring(1, 3).toLowerCase() + date.substring(3);
+                            }
```

Comment:
I think if you use the ExtractorUtils correctly, we could save this part.
What did you think?

Reply:
no, as the DateFormatter expects "Dec" and the text from the bank statement is "DEC"